src/NUnitFramework/framework/Assert.cs

```diff
@@ -318,8 +319,8 @@ private static void ReportFailure(string message)
             result.RecordAssertion(AssertionStatus.Failed, message, GetStackTrace());
             result.RecordTestCompletion();
 
-            // If we are outside any multiple assert block, then throw
-            if (TestExecutionContext.CurrentContext.MultipleAssertLevel == 0)
+            // If we are outside any multiple assert block or multiple asserts disabled, then throw
+            if (TestExecutionContext.CurrentContext.MultipleAssertLevel == 0 || (TestExecutionContext.CurrentContext.DisableMultipleAssertsUnderDebugger && Debugger.IsAttached))
```

Comment:
Check debugger status at exception throw location as user could connect/disconnect it during run